src/components/List.jsx

```diff
@@ -20,17 +21,66 @@ export const List = ({ token }) => {
     return timeDiff < 24;
   };
 
+  const calcDaysSince = (transactionDate) => {
+    const timeNow = moment().format();
+    const date1 = moment(timeNow, 'YYYYMMDD HH:mm:ss');
+    const date2 = moment(transactionDate, 'YYYYMMDD HH:mm:ss');
+    const timeDiff = date1.diff(date2, 'days');
+
+    return timeDiff;
+  };
+
+  const cleanDate = (date) => {
+    const lastPurchasedDate = moment(date).format('MMMM Do, YYYY');
+    return lastPurchasedDate;
+  };
+
   const updateDocument = async (id) => {
     const docRef = doc(db, token, id);
+    const document = await getDoc(docRef);
+    let docData;
+    if (document) {
+      docData = document.data();
+      console.log(docData);
+    }
+    const updateEstimate = () => {
+      // helper function to destructure fields for calculateEstimate
+      let prevEstimate;
+      const {
+        estimated_next_purchase,
+        days_since_last_transaction,
+        total_purchases,
+      } = docData;
+      // if no estimated_next_purchase exists, use 'undefined' in calculateEstimate (cannot save undefined fields in db)
+      estimated_next_purchase
+        ? (prevEstimate = estimated_next_purchase)
+        : (prevEstimate = undefined);
+
+      return calculateEstimate(
+        prevEstimate,
+        days_since_last_transaction,
+        total_purchases,
+      );
+    };
+    // this function runs when user selects item as purchased
     await updateDoc(docRef, {
-      purchased_date: moment().format(),
+      // if item has been purchased, use last_purchased_date, otherwise date_added
+      days_since_last_transaction: calcDaysSince(
+        docData.last_purchased_date || docData.date_added,
+      ),
+      // update last_purchased_date to current time
+      last_purchased_date: moment().format(),
+      // update total_purchases by 1
+      total_purchases: docData.total_purchases + 1,
+      // run updateEstimate helper to update estimated_next_purchase
+      estimated_next_purchase: updateEstimate(),
     });
   };
 
   const handleClick = (data, e) => {
     updateDocument(data.id);
   };
```

Comment:
I'm wondering if we pass in the entire docment object (called `data` here) instead of just its id to `updateDocument()`, could we set `docData` equal to `document.data()` without needing to use `getDoc()`? This would only remove a couple lines, so it's not a necessary change!

Reply:
Good suggestion. Looking at this one to implement.